client/src/app/components/common/new-file-modal.component.ts

```diff
@@ -113,15 +118,17 @@ export class NewFileModalComponent {
         Observable.of(1).delay(50).filter(_ => this.error === undefined).subscribe(_ => this.isCreatingFile = true);
 
         const fileType = this.fileTypeRadio.getSelectedValue();
-        let path       = new FileName(form.controls["filename"].value);
+        let fileName   = new FileName(form.controls["filename"].value);
 
         if (fileType === "workflow" || fileType === "command_line_tool") {
-            path = path.ensureExtension("json");
+            fileName = fileName.ensureExtension("json");
```

Comment:
In the case of creating a file that will be consumed only by javascript GUI? Absolutely none.

In the case of a user editing it by hand, many. JSON is a pain in the ass to write by hand. If our goal is to on board people to CWL, we don't want to force them to write JSON and all the double quotes, braces, brackets and commas that go with it.